e2e/wdio/headless/execute-leak.e2e.ts

```diff
@@ -0,0 +1,29 @@
+import { browser, expect } from '@wdio/globals'
+import logger from '@wdio/logger'
+
+const log = logger('test')
+
+function getHeapUsage() {
+    return process.memoryUsage().heapUsed / 1024 / 1024
+}
+
+describe('`execute` memory leak check', () => {
+    it('repeated identical `execute` script should not lead to memory leak', async () => {
+        await browser.url('https://example.com/')
+        // @ts-ignore
+        globalThis.gc()
+        const initialHeapUsage = getHeapUsage()
+        log.debug(`Initial heap usage ${initialHeapUsage.toFixed(2)} MiB`)
+        let heapUsage = 0
+        for (let i = 0; i < 50; i++) {
+            await browser.execute(`{ let foo = "${'a'.repeat(2_000_000)}"; }`)
+            heapUsage = getHeapUsage()
+            log.debug(`[${i.toString().padStart(2, '0')}] Heap usage ${heapUsage.toFixed(2)} MiB`)
+        }
+        // @ts-ignore
+        globalThis.gc()
+        const finalHeapUsage = getHeapUsage()
+        log.debug(`Final heap usage ${finalHeapUsage.toFixed(2)} MiB`)
+        await expect(finalHeapUsage).not.toBeGreaterThan(2 * initialHeapUsage)
```

Comment:
Can we add a comment to this assertion? Why do we expect it to only double in size?

Reply:
I also wonder if we can move this into a smoke test since we don't really care about if this is actually run in the browser. Smoke tests would run a bit more efficient and less flaky as there is no browser being run.